lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2391,6 +2391,17 @@ bool LLVMToSPIRV::transExecutionMode() {
           BM->addCapability(CapabilityFPGAKernelAttributesINTEL);
         }
       } break;
+
+      case spv::ExecutionModeDenormPreserve:
+      case spv::ExecutionModeDenormFlushToZero:
+      case spv::ExecutionModeSignedZeroInfNanPreserve:
+      case spv::ExecutionModeRoundingModeRTE:
+      case spv::ExecutionModeRoundingModeRTZ:{
+          unsigned TargetWidth;
+          N.get(TargetWidth);
+          BF->addExecutionMode(BM->add(new SPIRVExecutionMode(
```

Comment:
I don't see corresponding changes for the `SPIRVReader`, which means that this execution mode is going to be lost during reverse translation: are there plans to add SPIR-V -> LLVM IR support for this extension?

Reply:
The idea behind this commit is to introduce infrastructure, without any Reader/Writer part.
This particular change is also a part of infrastructure - before this piece of code will work, somebody has to add those Exec Modes in PreprocessMetadata.cpp
Until that, reader or writer will not even touch these exec modes: they just do not know from which structure in LLVM IR translate to ExecutionModeRoundingModeRTE, neither what to do with this mode in reader part.